src/node-jspm-jasmine.js

```diff
@@ -141,17 +179,79 @@ export function runTests(opts, errCallback = function() {}) {
 							importTheseTestFiles();
 						}
 					})
-					.catch(errCallback);
+					.catch((ex) => {
+						if (opts.coverage && opts.coverage.tempDirectory) {
+							// remove temporary directory
+							rimraf.sync(opts.coverage.tempDirectory)
+						}
+						errCallback(ex)
+					});
 				});
 			});
 		}
 	} catch(ex) {
-		removeTempFiles();
+		if (opts.coverage && opts.coverage.tempDirectory) {
+			// remove temporary directory
+			rimraf.sync(opts.coverage.tempDirectory)
+		}
 		errCallback(new Error(`Jasmine or Jspm may not be properly configured -- '${ex.toString()}'`));
 	}
 }
 
 function importTestFiles(SystemJS, jasmine, specDir, specFiles, coverage, errCallback) {
+
+	if (coverage) {
+		// jasmine will exit the process on its own, and since there doesn't appear
+		// to be a "onExit" hook, we're forced to intercept node's exit function and
+		// use as a "all done" jasmine hook.
+		// jasmine does offer a "addReporter" option, which contains a "jasmineDone"
+		// hook, however, it has no mechanism for waiting on async operations and
+		// async is needed for importing modules.
+		// Also, we should provide a "onExit" hook for the "programmatic" version,
+		// otherwise jasmine will abruptly kill the current running script.
+		const processExit = process.exit;
+		process.exit = function () {
```

Comment:
I pulled down your code and now it seems that it is impossible to kill the test process with a SIGINT or control+c. Seems like it's probably related to this change. We should look into finding a better way of detecting when jasmine is complete.